src/xtgeo/grid3d/_grdecl_grid.py

```diff
@@ -330,12 +330,16 @@ def _to_grdecl_file(self, filename):
             for kw, values in keywords:
                 if values is None:
                     continue
-                filestream.write(kw)
-                if values is not None:
-                    filestream.write("\n")
-                    for value in values:
-                        filestream.write(" ")
-                        filestream.write(str(value))
+                filestream.write(f"{kw}\n")
+                lcount = 0
+                for value in values:
+                    lcount += 1
+                    filestream.write(f" {value}")
+
+                    if kw in ("COORD", "ZCORN", "ACTNUM") and lcount >= 6:
```

Comment:
Maybe add a comment that 6 is assumed to be small enough so that characters pr line never exceed 128.

Reply:
OK